### PR TITLE
Validate version consistency

### PR DESCRIPTION
**Pull request by mtojek**
Issue: https://github.com/elastic/package-registry/issues/509

### CODE REVIEW

File: util/package.go [hunk 1]
```diff
@@ -316,9 +315,32 @@ func (p *Package) Validate() error {
...
+func (p *Package) validateVersionConsistency() error {
+	versionPackage, err := semver.Parse(p.Version)
+	if err != nil {
+		return errors.Wrap(err, "invalid version defined in manifest")
+	}
+
+	baseDir := filepath.Base(p.BasePath)
+	versionDir, err := semver.Parse(baseDir)
+	if err != nil {
+		return nil // package content is not rooted in version directory
```
Comment (ruflin) [inline util/package.go:335]: Do you expect us to support this long term or only at the moment?

File: util/package.go [hunk 2]
```diff
@@ -316,9 +315,32 @@ func (p *Package) Validate() error {
...
+	err = p.validateVersionConsistency()
+	if err != nil {
+		return errors.Wrap(err, "version in manifest file is not consistent with path")
+	}
+
 	return p.ValidateDatasets()
 }
 
+func (p *Package) validateVersionConsistency() error {
+	versionPackage, err := semver.Parse(p.Version)
```
Comment (ruflin) [inline util/package.go:327]: You could use this one here: https://github.com/elastic/package-registry/blob/d4a6d00d7eed510b8630273acdf061ec69a47b58/util/package.go#L38

File: main.go [hunk 3]
```diff
@@ -32,10 +32,9 @@ const (
 )
 
 var (
-	packagesBasePath string
```
Comment (ruflin) [inline main.go:35]: Seems like we don't use this anymore? My initial intention with this was to be able to pass it as a flag to to the docker container on which directories it should read.

File: util/package.go (hunk 1 above)
Comment (mtojek) [reply util/package.go:335]: I planned to leave it there long term. I put this check just in case.

File: util/package.go (hunk 2 above)
Comment (mtojek) [reply util/package.go:327]: No, I couldn't. The `Validate()` method is called by `Marshal()`, so the mentioned field is not filled yet.

File: main.go (hunk 3 above)
Comment (mtojek) [reply main.go:35]: I the next step I have a plan to support multiple sources of packages (exactly for the docker environment). I think we can leave it as is.

**Pull request merged**